lib/matplotlib/axes/_axes.py

```diff
@@ -4117,21 +4121,38 @@ def dopatch(xs, ys, **kwargs):
                     flier_x, flier_y, **final_flierprops
                 ))
 
-        # fix our axes/ticks up a little
-        if vert:
-            setticks = self.set_xticks
-            setlim = self.set_xlim
-            setlabels = self.set_xticklabels
-        else:
-            setticks = self.set_yticks
-            setlim = self.set_ylim
-            setlabels = self.set_yticklabels
-
-        if manage_xticks:
-            newlimits = min(positions) - 0.5, max(positions) + 0.5
-            setlim(newlimits)
-            setticks(positions)
-            setlabels(datalabels)
+        if manage_ticks:
+            axis_name = "x" if vert else "y"
+            interval = getattr(self.dataLim, f"interval{axis_name}")
+            axis = getattr(self, f"{axis_name}axis")
+            positions = axis.convert_units(positions)
+            # The 0.5 additional padding ensures reasonable-looking boxes
+            # even when drawing a single box.  We set the sticky edge to
+            # prevent margins expansion, in order to match old behavior (back
+            # when separate calls to boxplot() would completely reset the axis
+            # limits regardless of what was drawn before).  The sticky edges
+            # are attached to the median lines, as they are always present.
+            interval[:] = (min(interval[0], min(positions) - .5),
+                           max(interval[1], max(positions) + .5))
+            for median, position in zip(medians, positions):
```

Comment:
Does this interval/sticky_edes stuff keep the exact appearance for single `boxplot`-call figures? Probably, since the tests pass - OTOH not sure how good our test coverage is.

Reply:
Yes, this is tested (initially the tests failed, then I initially added some additional kwarg to force using the old method, then I realized that sticky_edges could be used for that purpose).